src/staticanalysis/bot/static_analysis_bot/infer/infer.py

```diff
@@ -0,0 +1,196 @@
+        self.revision = revision
+
+        # Run all files in a single command
+        # through mach static-analysis
+        cmd = [
+            'gecko-env',
+            './mach', '--log-no-times', 'static-analysis', 'check-java'
+        ] + list(revision.files)
+        logger.info('Running static-analysis', cmd=' '.join(cmd))
+
+        # Run command
+        try:
+            infer_output = subprocess.check_output(cmd, cwd=settings.repo_dir)
+        except subprocess.CalledProcessError as e:
+            logger.error('Mach static analysis failed: {}'.format(e.output))
+            raise
+
+        import json
+        report_file = os.path.join(settings.repo_dir, 'infer-out', 'report.json')
+        infer_output = json.load(open(report_file))
+        # Dump raw clang-tidy output as a Taskcluster artifact (for debugging)
+        infer_output_path = os.path.join(
+            settings.taskcluster_results_dir,
+            '{}-infer.txt'.format(repr(revision)),
+        )
+        with open(infer_output_path, 'w') as f:
+            f.write(json.dumps(infer_output, indent=2))
+        issues = self.parse_issues(infer_output, revision)
+        # Report stats for these issues
+        stats.report_issues('infer', issues)
+        return issues
+
+    def parse_issues(self, infer_output, revision):
+        '''
+        Parse infer output into structured issues
+        '''
+        if not infer_output:
+            logger.info('Infer could not generate any output.')
+            return []
+        # check if json is empty
+        nb_warnings = len(infer_output)
+        if not nb_warnings:
+            logger.info('Mach static analysis did not find any issue')
+            return []
+        logger.info('Mach static analysis found some issues', nb=nb_warnings)
+        return [InferIssue(issue, revision) for issue in infer_output]
+
+
+class InferIssue(Issue):
+    '''
+    An issue reported by infer
+    '''
+    def __init__(self, entry, revision):
+        assert isinstance(entry, dict)
+        assert not settings.repo_dir.endswith('/')
+        self.revision = revision
+        self.path = entry['file']
+        self.line = entry['line']
+        self.column = entry['column']
+        self.bug_type = entry['bug_type']
+        self.kind = entry['kind']
+        self.message = entry['qualifier']
+        self.body = None
+        self.nb_lines = 1
+
+    def __str__(self):
+        return '[{}] {} {}:{}'.format(self.bug_type, self.path,
+                                      self.line, self.column)
+
+    def build_extra_identifiers(self):
+        '''
+        Used to compare with same-class issues
+        '''
+        return {
+            'bug_type': self.bug_type,
+            'kind': self.kind,
+            'column': self.column,
+        }
+
+    def is_problem(self):
+        return self.kind in ('ERROR')
+
+    def validates(self):
+        '''
+        Publish infer issues all the time
+        '''
+        return True
```

Comment:
Are you sure about this ? Do we want to publish every issue on Phabricator (that's what will happend on prod here)? 

Is there some formatting only warnings ? We may want to skip those for now...

Reply:
`infer` does not look for formatting issues, only logical ones.